src/basset.cpp

```diff
@@ -29,6 +29,46 @@ using std::unordered_set;
 using std::vector;
 using std::literals::string_literals::operator""s;
 
+string json_escape(const string &str) {
+  static const char hex[] = "0123456789abcdef";
+  std::string s;
+
+  for (const auto c : str) {
+    if (c == '\\' || c == '"') {
+      s.push_back('\\');
+      s.push_back(c);
+      continue;
+    }
+    if (c < 0x20) {
```

Comment:
It is a bit odd, that here you use `continue`, but later (line 64) you use `else`.

I think it would be nice to convert the body of the `for` into a nice uniform `if ... else if ... else ...` construct:
```suggestion
    } else if (c < 0x20) {
```